lib/src/native/realm_core.dart

```diff
@@ -1104,6 +1104,82 @@ class _RealmCore {
           "Switch user failed");
     });
   }
+
+  String userGetCustomData(User user) {
+    final customDataPtr = _realmLib.invokeGetPointer(() => _realmLib.realm_user_get_custom_data(user.handle._pointer));
+    try {
+      final customData = customDataPtr.cast<Utf8>().toDartString();
+      return customData;
+    } finally {
+      _realmLib.realm_free(customDataPtr.cast());
+    }
+  }
+
+  Future<void> userRefreshCustomData(App app, User user) {
+    final completer = Completer<void>();
+    _realmLib.invokeGetBool(
+        () => _realmLib.realm_app_refresh_custom_data(
+              app.handle._pointer,
+              user.handle._pointer,
+              Pointer.fromFunction(void_completion_callback),
+              completer.toPersistentHandle(),
+              _deletePersistentHandleFuncPtr,
+            ),
+        "Refresh custom data failed");
+    return completer.future;
+  }
+
+  Future<UserHandle> userLinkCredentials(App app, User user, Credentials credentials) {
+    final completer = Completer<UserHandle>();
+    _realmLib.invokeGetBool(
+        () => _realmLib.realm_app_link_user(
+              app.handle._pointer,
+              user.handle._pointer,
+              credentials.handle._pointer,
+              Pointer.fromFunction(_app_user_completion_callback),
+              completer.toPersistentHandle(),
+              _deletePersistentHandleFuncPtr,
+            ),
+        "Link credentials failed");
+    return completer.future;
+  }
+
+  int userGetState(User user) {
+    return _realmLib.realm_user_get_state(user.handle._pointer);
+  }
+
+  UserIdentity userGetIdentity(User user) {
```

Comment:
This should return just a string, no need to convert it to a `UserIdentity` object.

Reply:
I know .NET has this as a string. My thinking here it is more on par with the other API. So we have a getter returning a single `UserIdentity` and one that returns multiple ones as List<UserIdentity>. The properties are named 'identity' and `identities` which kind of implies more aligned return type.

Reply:
Those are different though. The User has an Id which is unique to that user and then they have one or more identities each with their unique ids. If I have a user that logged in with email/password auth, I could have something like:

```
{
  id: "123",
  identities: [
    { id: "abc", provider: "password" }
  ]
}
```

I think it would be weird if we return an identity here and `user.identity != user.identitites.single`.

Reply:
got it will fix